### ISSUE TITLE
Render notification using Action Cable

### ISSUE THREAD

**Issue by juliolugo96** [simukappu/activity_notification#101]
Hi, first of all, what a great gem. Amazing, sincerely.

I'm trying to show a dropdown list with all notifications I receive in real time, with all their links and styles. Basically, **common functionalities but using WebSockets**.

Initially, I'm using ActionCable and I've been able to solve earlier issues using what was discussed here (#94). However, I couldn't figure out how to call `render_notification` method or something similar just after a notification is triggered and received by my Notifications subscriber in JavaScript.

I'd like to **receive the result of calling render_notification** (an HTML) as a string format directly from server side, and then **prepend it to my dropdown** (I've been trying using JavaScript).

Is there any way to acomplish this? Thank you!

Edit: I'd like to add one interesting behaviour I found:

```ruby
# app/jobs/notification_broadcast_job.rb

  def self.perform(notification)
    ActionCable.server.broadcast target_channel(notification),
                                 notification: notification
  end

# If I try something like

 def self.perform(notification)
    ActionCable.server.broadcast target_channel(notification),
                                 notification: notification,
                                 render: notification.render(notification, {})
 end

# I obtain an error indicating that notification is a Hash, but clearly is a Notification instance

```

**Comment by simukappu** [comment]
Sorry for my late response.
For example, you can return the rendered notification to ActionCable subscriber like this:
```ruby
def self.perform(notification)
  ActionCable.server.broadcast target_channel(notification),
                               notification: notification,
                               render: render(ActivityNotification::NotificationsController.renderer, fallback: :default))
end
```

Now, I'm developing reference implementation included in *activity_notification* gem working with ActionCable for the next version. If you provide any feedback about this use case, I would appreciate it.

**Comment by juliolugo96** [comment]
Hi, thanks for your response. I was able to solve this (at least in a development environment) using the following code:

``` ruby

# app/jobs/notification_broadcast_job.rb

class NotificationBroadcastJob < ApplicationJob
  def self.perform(notification)
    ActionCable.server.broadcast target_channel(notification),
                                 notification: notification,
                                 view: render_notification(notification)
  end

  private

  def self.target_channel(notification)
    "notifications_channel_user_#{notification.target_id}"
  end

  def self.render_notification(notification)
# I'm working with different notifications type
    case notification.notifiable_type
    when 'Card'
      ApplicationController.renderer.render partial: 'activity_notification/notifications/default/card/create',
                                            locals: { notification: notification }
    when 'CardComment'
      ApplicationController.renderer.render partial: 'activity_notification/notifications/default/card_comment/create',
                                            locals: { notification: notification }
    when 'CardsMembership'
      ApplicationController.renderer.render partial: 'activity_notification/notifications/default/cards_membership/create',
   end
  end
end
```

I'm going to implement your approach also and share obtained results here.

**Comment by simukappu** [comment]
We've published v2.0.0 which includes new push notification function with Action Cable.
https://rubygems.org/gems/activity_notification/versions/2.0.0

See [Push notification with Action Cable](https://github.com/simukappu/activity_notification#push-notification-with-action-cable) for more details.
Please try v2 gem and give us any feedback if it works for you!
